examples/CMakeLists.txt

```diff
@@ -60,6 +61,24 @@ blt_add_executable(
   DEPENDS_ON ${example_depends})
 list(APPEND umpire_examples strategy_example)
 
+if (UMPIRE_ENABLE_CUDA)
+blt_add_executable(
+  NAME rap_example
+  SOURCES rap_example.cpp
+  DEPENDS_ON ${example_depends} cuda)
```

Comment:
Doesn't the umpire CMake target already have cuda or blt::hip baked in?

Reply:
Well, I think my reasoning here was that we always need the ResourceAwarePool, but if CUDA is enabled, this example should use CUDA - Or, if Hip is enabled, it should use Hip - Or, if neither are enabled, then it should just use the Host. Is there a better way to set that up?
Currently I have
```
 64 if (UMPIRE_ENABLE_CUDA)
 65 blt_add_executable(
 66   NAME rap_example
 67   SOURCES rap_example.cpp
 68   DEPENDS_ON ${example_depends} cuda)
 69 elseif (UMPIRE_ENABLE_HIP)
 70 blt_add_executable(
 71   NAME rap_example
 72   SOURCES rap_example.cpp
 73   DEPENDS_ON ${example_depends} blt::hip)
 74 else()
 75 blt_add_executable(
 76   NAME rap_example
 77   SOURCES rap_example.cpp
 78   DEPENDS_ON ${example_depends})
 79 endif()
 80 list(APPEND umpire_examples rap_example)
```
I have another idea on how to do this, but I am not sure if there a way to ask blt/cmake "If you don't already have this exe in the list of executables, then...." ?